// @floating-ui/react/dist/floating-ui.react.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AlignedPlacement", {
  enumerable: true,
  get: function () {
    return _reactDom.AlignedPlacement;
  }
});
Object.defineProperty(exports, "Alignment", {
  enumerable: true,
  get: function () {
    return _reactDom.Alignment;
  }
});
Object.defineProperty(exports, "ArrowOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.ArrowOptions;
  }
});
Object.defineProperty(exports, "AutoPlacementOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.AutoPlacementOptions;
  }
});
Object.defineProperty(exports, "AutoUpdateOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.AutoUpdateOptions;
  }
});
Object.defineProperty(exports, "Axis", {
  enumerable: true,
  get: function () {
    return _reactDom.Axis;
  }
});
Object.defineProperty(exports, "Boundary", {
  enumerable: true,
  get: function () {
    return _reactDom.Boundary;
  }
});
Object.defineProperty(exports, "ClientRectObject", {
  enumerable: true,
  get: function () {
    return _reactDom.ClientRectObject;
  }
});
Object.defineProperty(exports, "ComputePositionConfig", {
  enumerable: true,
  get: function () {
    return _reactDom.ComputePositionConfig;
  }
});
Object.defineProperty(exports, "ComputePositionReturn", {
  enumerable: true,
  get: function () {
    return _reactDom.ComputePositionReturn;
  }
});
Object.defineProperty(exports, "Coords", {
  enumerable: true,
  get: function () {
    return _reactDom.Coords;
  }
});
Object.defineProperty(exports, "DetectOverflowOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.DetectOverflowOptions;
  }
});
Object.defineProperty(exports, "Dimensions", {
  enumerable: true,
  get: function () {
    return _reactDom.Dimensions;
  }
});
Object.defineProperty(exports, "ElementContext", {
  enumerable: true,
  get: function () {
    return _reactDom.ElementContext;
  }
});
Object.defineProperty(exports, "ElementRects", {
  enumerable: true,
  get: function () {
    return _reactDom.ElementRects;
  }
});
Object.defineProperty(exports, "Elements", {
  enumerable: true,
  get: function () {
    return _reactDom.Elements;
  }
});
Object.defineProperty(exports, "FlipOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.FlipOptions;
  }
});
Object.defineProperty(exports, "FloatingElement", {
  enumerable: true,
  get: function () {
    return _reactDom.FloatingElement;
  }
});
Object.defineProperty(exports, "HideOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.HideOptions;
  }
});
Object.defineProperty(exports, "InlineOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.InlineOptions;
  }
});
Object.defineProperty(exports, "Length", {
  enumerable: true,
  get: function () {
    return _reactDom.Length;
  }
});
Object.defineProperty(exports, "Middleware", {
  enumerable: true,
  get: function () {
    return _reactDom.Middleware;
  }
});
Object.defineProperty(exports, "MiddlewareArguments", {
  enumerable: true,
  get: function () {
    return _reactDom.MiddlewareArguments;
  }
});
Object.defineProperty(exports, "MiddlewareData", {
  enumerable: true,
  get: function () {
    return _reactDom.MiddlewareData;
  }
});
Object.defineProperty(exports, "MiddlewareReturn", {
  enumerable: true,
  get: function () {
    return _reactDom.MiddlewareReturn;
  }
});
Object.defineProperty(exports, "MiddlewareState", {
  enumerable: true,
  get: function () {
    return _reactDom.MiddlewareState;
  }
});
Object.defineProperty(exports, "NodeScroll", {
  enumerable: true,
  get: function () {
    return _reactDom.NodeScroll;
  }
});
Object.defineProperty(exports, "OffsetOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.OffsetOptions;
  }
});
Object.defineProperty(exports, "Padding", {
  enumerable: true,
  get: function () {
    return _reactDom.Padding;
  }
});
Object.defineProperty(exports, "Placement", {
  enumerable: true,
  get: function () {
    return _reactDom.Placement;
  }
});
Object.defineProperty(exports, "Platform", {
  enumerable: true,
  get: function () {
    return _reactDom.Platform;
  }
});
Object.defineProperty(exports, "Rect", {
  enumerable: true,
  get: function () {
    return _reactDom.Rect;
  }
});
Object.defineProperty(exports, "ReferenceElement", {
  enumerable: true,
  get: function () {
    return _reactDom.ReferenceElement;
  }
});
Object.defineProperty(exports, "RootBoundary", {
  enumerable: true,
  get: function () {
    return _reactDom.RootBoundary;
  }
});
Object.defineProperty(exports, "ShiftOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.ShiftOptions;
  }
});
Object.defineProperty(exports, "Side", {
  enumerable: true,
  get: function () {
    return _reactDom.Side;
  }
});
Object.defineProperty(exports, "SideObject", {
  enumerable: true,
  get: function () {
    return _reactDom.SideObject;
  }
});
Object.defineProperty(exports, "SizeOptions", {
  enumerable: true,
  get: function () {
    return _reactDom.SizeOptions;
  }
});
Object.defineProperty(exports, "Strategy", {
  enumerable: true,
  get: function () {
    return _reactDom.Strategy;
  }
});
Object.defineProperty(exports, "VirtualElement", {
  enumerable: true,
  get: function () {
    return _reactDom.VirtualElement;
  }
});
Object.defineProperty(exports, "arrow", {
  enumerable: true,
  get: function () {
    return _reactDom.arrow;
  }
});
Object.defineProperty(exports, "autoPlacement", {
  enumerable: true,
  get: function () {
    return _reactDom.autoPlacement;
  }
});
Object.defineProperty(exports, "autoUpdate", {
  enumerable: true,
  get: function () {
    return _reactDom.autoUpdate;
  }
});
Object.defineProperty(exports, "computePosition", {
  enumerable: true,
  get: function () {
    return _reactDom.computePosition;
  }
});
Object.defineProperty(exports, "detectOverflow", {
  enumerable: true,
  get: function () {
    return _reactDom.detectOverflow;
  }
});
Object.defineProperty(exports, "flip", {
  enumerable: true,
  get: function () {
    return _reactDom.flip;
  }
});
Object.defineProperty(exports, "getOverflowAncestors", {
  enumerable: true,
  get: function () {
    return _reactDom.getOverflowAncestors;
  }
});
Object.defineProperty(exports, "hide", {
  enumerable: true,
  get: function () {
    return _reactDom.hide;
  }
});
Object.defineProperty(exports, "inline", {
  enumerable: true,
  get: function () {
    return _reactDom.inline;
  }
});
Object.defineProperty(exports, "limitShift", {
  enumerable: true,
  get: function () {
    return _reactDom.limitShift;
  }
});
Object.defineProperty(exports, "offset", {
  enumerable: true,
  get: function () {
    return _reactDom.offset;
  }
});
Object.defineProperty(exports, "platform", {
  enumerable: true,
  get: function () {
    return _reactDom.platform;
  }
});
Object.defineProperty(exports, "shift", {
  enumerable: true,
  get: function () {
    return _reactDom.shift;
  }
});
Object.defineProperty(exports, "size", {
  enumerable: true,
  get: function () {
    return _reactDom.size;
  }
});
var _reactDom = require("@floating-ui/react-dom");
