// @mantine/hooks/lib/utils/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "clamp", {
  enumerable: true,
  get: function () {
    return _clamp.clamp;
  }
});
Object.defineProperty(exports, "lowerFirst", {
  enumerable: true,
  get: function () {
    return _lowerFirst.lowerFirst;
  }
});
Object.defineProperty(exports, "randomId", {
  enumerable: true,
  get: function () {
    return _randomId.randomId;
  }
});
Object.defineProperty(exports, "range", {
  enumerable: true,
  get: function () {
    return _range.range;
  }
});
Object.defineProperty(exports, "shallowEqual", {
  enumerable: true,
  get: function () {
    return _shallowEqual.shallowEqual;
  }
});
Object.defineProperty(exports, "upperFirst", {
  enumerable: true,
  get: function () {
    return _upperFirst.upperFirst;
  }
});
var _clamp = require("./clamp/clamp");
var _lowerFirst = require("./lower-first/lower-first");
var _randomId = require("./random-id/random-id");
var _range = require("./range/range");
var _shallowEqual = require("./shallow-equal/shallow-equal");
var _upperFirst = require("./upper-first/upper-first");
