// @mantine/core/lib/core/utils/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "camelToKebabCase", {
  enumerable: true,
  get: function () {
    return _camelToKebabCase.camelToKebabCase;
  }
});
Object.defineProperty(exports, "closeOnEscape", {
  enumerable: true,
  get: function () {
    return _closeOnEscape.closeOnEscape;
  }
});
Object.defineProperty(exports, "createEventHandler", {
  enumerable: true,
  get: function () {
    return _createEventHandler.createEventHandler;
  }
});
Object.defineProperty(exports, "createOptionalContext", {
  enumerable: true,
  get: function () {
    return _createOptionalContext.createOptionalContext;
  }
});
Object.defineProperty(exports, "createSafeContext", {
  enumerable: true,
  get: function () {
    return _createSafeContext.createSafeContext;
  }
});
Object.defineProperty(exports, "createScopedKeydownHandler", {
  enumerable: true,
  get: function () {
    return _createScopedKeydownHandler.createScopedKeydownHandler;
  }
});
Object.defineProperty(exports, "createUseExternalEvents", {
  enumerable: true,
  get: function () {
    return _createUseExternalEvents.createUseExternalEvents;
  }
});
Object.defineProperty(exports, "deepMerge", {
  enumerable: true,
  get: function () {
    return _deepMerge.deepMerge;
  }
});
Object.defineProperty(exports, "em", {
  enumerable: true,
  get: function () {
    return _unitsConverters.em;
  }
});
Object.defineProperty(exports, "filterProps", {
  enumerable: true,
  get: function () {
    return _filterProps.filterProps;
  }
});
Object.defineProperty(exports, "findElementAncestor", {
  enumerable: true,
  get: function () {
    return _findElementAncestor.findElementAncestor;
  }
});
Object.defineProperty(exports, "getBaseValue", {
  enumerable: true,
  get: function () {
    return _getBaseValue.getBaseValue;
  }
});
Object.defineProperty(exports, "getBreakpointValue", {
  enumerable: true,
  get: function () {
    return _getBreakpointValue.getBreakpointValue;
  }
});
Object.defineProperty(exports, "getContextItemIndex", {
  enumerable: true,
  get: function () {
    return _getContextItemIndex.getContextItemIndex;
  }
});
Object.defineProperty(exports, "getDefaultZIndex", {
  enumerable: true,
  get: function () {
    return _getDefaultZIndex.getDefaultZIndex;
  }
});
Object.defineProperty(exports, "getEnv", {
  enumerable: true,
  get: function () {
    return _getEnv.getEnv;
  }
});
Object.defineProperty(exports, "getFontSize", {
  enumerable: true,
  get: function () {
    return _getSize.getFontSize;
  }
});
Object.defineProperty(exports, "getLineHeight", {
  enumerable: true,
  get: function () {
    return _getSize.getLineHeight;
  }
});
Object.defineProperty(exports, "getRadius", {
  enumerable: true,
  get: function () {
    return _getSize.getRadius;
  }
});
Object.defineProperty(exports, "getSafeId", {
  enumerable: true,
  get: function () {
    return _getSafeId.getSafeId;
  }
});
Object.defineProperty(exports, "getShadow", {
  enumerable: true,
  get: function () {
    return _getSize.getShadow;
  }
});
Object.defineProperty(exports, "getSize", {
  enumerable: true,
  get: function () {
    return _getSize.getSize;
  }
});
Object.defineProperty(exports, "getSortedBreakpoints", {
  enumerable: true,
  get: function () {
    return _getSortedBreakpoints.getSortedBreakpoints;
  }
});
Object.defineProperty(exports, "getSpacing", {
  enumerable: true,
  get: function () {
    return _getSize.getSpacing;
  }
});
Object.defineProperty(exports, "isElement", {
  enumerable: true,
  get: function () {
    return _isElement.isElement;
  }
});
Object.defineProperty(exports, "isNumberLike", {
  enumerable: true,
  get: function () {
    return _isNumberLike.isNumberLike;
  }
});
Object.defineProperty(exports, "keys", {
  enumerable: true,
  get: function () {
    return _keys.keys;
  }
});
Object.defineProperty(exports, "noop", {
  enumerable: true,
  get: function () {
    return _noop.noop;
  }
});
Object.defineProperty(exports, "px", {
  enumerable: true,
  get: function () {
    return _unitsConverters.px;
  }
});
Object.defineProperty(exports, "rem", {
  enumerable: true,
  get: function () {
    return _unitsConverters.rem;
  }
});
Object.defineProperty(exports, "useHovered", {
  enumerable: true,
  get: function () {
    return _useHovered.useHovered;
  }
});
var _keys = require("./keys/keys");
var _deepMerge = require("./deep-merge/deep-merge");
var _camelToKebabCase = require("./camel-to-kebab-case/camel-to-kebab-case");
var _unitsConverters = require("./units-converters");
var _filterProps = require("./filter-props/filter-props");
var _isNumberLike = require("./is-number-like/is-number-like");
var _isElement = require("./is-element/is-element");
var _createSafeContext = require("./create-safe-context/create-safe-context");
var _createOptionalContext = require("./create-optional-context/create-optional-context");
var _getSafeId = require("./get-safe-id/get-safe-id");
var _createScopedKeydownHandler = require("./create-scoped-keydown-handler/create-scoped-keydown-handler");
var _findElementAncestor = require("./find-element-ancestor/find-element-ancestor");
var _getDefaultZIndex = require("./get-default-z-index/get-default-z-index");
var _closeOnEscape = require("./close-on-escape/close-on-escape");
var _noop = require("./noop/noop");
var _getSize = require("./get-size/get-size");
var _createEventHandler = require("./create-event-handler/create-event-handler");
var _getBreakpointValue = require("./get-breakpoint-value/get-breakpoint-value");
var _getSortedBreakpoints = require("./get-sorted-breakpoints/get-sorted-breakpoints");
var _getBaseValue = require("./get-base-value/get-base-value");
var _getContextItemIndex = require("./get-context-item-index/get-context-item-index");
var _useHovered = require("./use-hovered/use-hovered");
var _createUseExternalEvents = require("./create-use-external-events/create-use-external-events");
var _getEnv = require("./get-env/get-env");
