// @mantine/core/lib/components/Drawer/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Drawer", {
  enumerable: true,
  get: function () {
    return _Drawer.Drawer;
  }
});
Object.defineProperty(exports, "DrawerBody", {
  enumerable: true,
  get: function () {
    return _DrawerBody.DrawerBody;
  }
});
Object.defineProperty(exports, "DrawerCloseButton", {
  enumerable: true,
  get: function () {
    return _DrawerCloseButton.DrawerCloseButton;
  }
});
Object.defineProperty(exports, "DrawerContent", {
  enumerable: true,
  get: function () {
    return _DrawerContent.DrawerContent;
  }
});
Object.defineProperty(exports, "DrawerHeader", {
  enumerable: true,
  get: function () {
    return _DrawerHeader.DrawerHeader;
  }
});
Object.defineProperty(exports, "DrawerOverlay", {
  enumerable: true,
  get: function () {
    return _DrawerOverlay.DrawerOverlay;
  }
});
Object.defineProperty(exports, "DrawerRoot", {
  enumerable: true,
  get: function () {
    return _DrawerRoot.DrawerRoot;
  }
});
Object.defineProperty(exports, "DrawerTitle", {
  enumerable: true,
  get: function () {
    return _DrawerTitle.DrawerTitle;
  }
});
var _Drawer = require("./Drawer");
var _DrawerRoot = require("./DrawerRoot");
var _DrawerBody = require("./DrawerBody");
var _DrawerCloseButton = require("./DrawerCloseButton");
var _DrawerContent = require("./DrawerContent");
var _DrawerHeader = require("./DrawerHeader");
var _DrawerOverlay = require("./DrawerOverlay");
var _DrawerTitle = require("./DrawerTitle");
