// @mantine/core/lib/components/AppShell/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AppShell", {
  enumerable: true,
  get: function () {
    return _AppShell.AppShell;
  }
});
Object.defineProperty(exports, "AppShellAside", {
  enumerable: true,
  get: function () {
    return _AppShellAside.AppShellAside;
  }
});
Object.defineProperty(exports, "AppShellFooter", {
  enumerable: true,
  get: function () {
    return _AppShellFooter.AppShellFooter;
  }
});
Object.defineProperty(exports, "AppShellHeader", {
  enumerable: true,
  get: function () {
    return _AppShellHeader.AppShellHeader;
  }
});
Object.defineProperty(exports, "AppShellMain", {
  enumerable: true,
  get: function () {
    return _AppShellMain.AppShellMain;
  }
});
Object.defineProperty(exports, "AppShellNavbar", {
  enumerable: true,
  get: function () {
    return _AppShellNavbar.AppShellNavbar;
  }
});
Object.defineProperty(exports, "AppShellSection", {
  enumerable: true,
  get: function () {
    return _AppShellSection.AppShellSection;
  }
});
var _AppShell = require("./AppShell");
var _AppShellAside = require("./AppShellAside/AppShellAside");
var _AppShellFooter = require("./AppShellFooter/AppShellFooter");
var _AppShellHeader = require("./AppShellHeader/AppShellHeader");
var _AppShellNavbar = require("./AppShellNavbar/AppShellNavbar");
var _AppShellSection = require("./AppShellSection/AppShellSection");
var _AppShellMain = require("./AppShellMain/AppShellMain");
