// @mantine/hooks/lib/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  useCallbackRef: true,
  useDebouncedCallback: true,
  useClickOutside: true,
  useClipboard: true,
  useColorScheme: true,
  useCounter: true,
  useDebouncedState: true,
  useDebouncedValue: true,
  useDocumentTitle: true,
  useDocumentVisibility: true,
  useFocusReturn: true,
  useDidUpdate: true,
  useFocusTrap: true,
  useForceUpdate: true,
  useId: true,
  useIdle: true,
  useInterval: true,
  useIsomorphicEffect: true,
  useListState: true,
  useLocalStorage: true,
  readLocalStorageValue: true,
  useSessionStorage: true,
  readSessionStorageValue: true,
  useMediaQuery: true,
  useMergedRef: true,
  mergeRefs: true,
  assignRef: true,
  useMouse: true,
  useMove: true,
  clampUseMovePosition: true,
  usePagination: true,
  useQueue: true,
  usePageLeave: true,
  useReducedMotion: true,
  useScrollIntoView: true,
  useResizeObserver: true,
  useElementSize: true,
  useShallowEffect: true,
  useToggle: true,
  useUncontrolled: true,
  useViewportSize: true,
  useWindowEvent: true,
  useWindowScroll: true,
  useIntersection: true,
  useHash: true,
  useHotkeys: true,
  getHotkeyHandler: true,
  useFullscreen: true,
  useLogger: true,
  useHover: true,
  useValidatedState: true,
  useOs: true,
  useSetState: true,
  useInputState: true,
  useEventListener: true,
  useDisclosure: true,
  useFocusWithin: true,
  useNetwork: true,
  useTimeout: true,
  useTextSelection: true,
  usePrevious: true,
  useFavicon: true,
  useHeadroom: true,
  useEyeDropper: true,
  useInViewport: true,
  useMutationObserver: true,
  useMounted: true,
  useStateHistory: true,
  useMap: true,
  useSet: true,
  useThrottledCallback: true,
  useThrottledState: true,
  useThrottledValue: true,
  useIsFirstRender: true,
  useOrientation: true,
  useFetch: true
};
Object.defineProperty(exports, "assignRef", {
  enumerable: true,
  get: function () {
    return _useMergedRef.assignRef;
  }
});
Object.defineProperty(exports, "clampUseMovePosition", {
  enumerable: true,
  get: function () {
    return _useMove.clampUseMovePosition;
  }
});
Object.defineProperty(exports, "getHotkeyHandler", {
  enumerable: true,
  get: function () {
    return _useHotkeys.getHotkeyHandler;
  }
});
Object.defineProperty(exports, "mergeRefs", {
  enumerable: true,
  get: function () {
    return _useMergedRef.mergeRefs;
  }
});
Object.defineProperty(exports, "readLocalStorageValue", {
  enumerable: true,
  get: function () {
    return _useLocalStorage.readLocalStorageValue;
  }
});
Object.defineProperty(exports, "readSessionStorageValue", {
  enumerable: true,
  get: function () {
    return _useSessionStorage.readSessionStorageValue;
  }
});
Object.defineProperty(exports, "useCallbackRef", {
  enumerable: true,
  get: function () {
    return _useCallbackRef.useCallbackRef;
  }
});
Object.defineProperty(exports, "useClickOutside", {
  enumerable: true,
  get: function () {
    return _useClickOutside.useClickOutside;
  }
});
Object.defineProperty(exports, "useClipboard", {
  enumerable: true,
  get: function () {
    return _useClipboard.useClipboard;
  }
});
Object.defineProperty(exports, "useColorScheme", {
  enumerable: true,
  get: function () {
    return _useColorScheme.useColorScheme;
  }
});
Object.defineProperty(exports, "useCounter", {
  enumerable: true,
  get: function () {
    return _useCounter.useCounter;
  }
});
Object.defineProperty(exports, "useDebouncedCallback", {
  enumerable: true,
  get: function () {
    return _useDebouncedCallback.useDebouncedCallback;
  }
});
Object.defineProperty(exports, "useDebouncedState", {
  enumerable: true,
  get: function () {
    return _useDebouncedState.useDebouncedState;
  }
});
Object.defineProperty(exports, "useDebouncedValue", {
  enumerable: true,
  get: function () {
    return _useDebouncedValue.useDebouncedValue;
  }
});
Object.defineProperty(exports, "useDidUpdate", {
  enumerable: true,
  get: function () {
    return _useDidUpdate.useDidUpdate;
  }
});
Object.defineProperty(exports, "useDisclosure", {
  enumerable: true,
  get: function () {
    return _useDisclosure.useDisclosure;
  }
});
Object.defineProperty(exports, "useDocumentTitle", {
  enumerable: true,
  get: function () {
    return _useDocumentTitle.useDocumentTitle;
  }
});
Object.defineProperty(exports, "useDocumentVisibility", {
  enumerable: true,
  get: function () {
    return _useDocumentVisibility.useDocumentVisibility;
  }
});
Object.defineProperty(exports, "useElementSize", {
  enumerable: true,
  get: function () {
    return _useResizeObserver.useElementSize;
  }
});
Object.defineProperty(exports, "useEventListener", {
  enumerable: true,
  get: function () {
    return _useEventListener.useEventListener;
  }
});
Object.defineProperty(exports, "useEyeDropper", {
  enumerable: true,
  get: function () {
    return _useEyeDropper.useEyeDropper;
  }
});
Object.defineProperty(exports, "useFavicon", {
  enumerable: true,
  get: function () {
    return _useFavicon.useFavicon;
  }
});
Object.defineProperty(exports, "useFetch", {
  enumerable: true,
  get: function () {
    return _useFetch.useFetch;
  }
});
Object.defineProperty(exports, "useFocusReturn", {
  enumerable: true,
  get: function () {
    return _useFocusReturn.useFocusReturn;
  }
});
Object.defineProperty(exports, "useFocusTrap", {
  enumerable: true,
  get: function () {
    return _useFocusTrap.useFocusTrap;
  }
});
Object.defineProperty(exports, "useFocusWithin", {
  enumerable: true,
  get: function () {
    return _useFocusWithin.useFocusWithin;
  }
});
Object.defineProperty(exports, "useForceUpdate", {
  enumerable: true,
  get: function () {
    return _useForceUpdate.useForceUpdate;
  }
});
Object.defineProperty(exports, "useFullscreen", {
  enumerable: true,
  get: function () {
    return _useFullscreen.useFullscreen;
  }
});
Object.defineProperty(exports, "useHash", {
  enumerable: true,
  get: function () {
    return _useHash.useHash;
  }
});
Object.defineProperty(exports, "useHeadroom", {
  enumerable: true,
  get: function () {
    return _useHeadroom.useHeadroom;
  }
});
Object.defineProperty(exports, "useHotkeys", {
  enumerable: true,
  get: function () {
    return _useHotkeys.useHotkeys;
  }
});
Object.defineProperty(exports, "useHover", {
  enumerable: true,
  get: function () {
    return _useHover.useHover;
  }
});
Object.defineProperty(exports, "useId", {
  enumerable: true,
  get: function () {
    return _useId.useId;
  }
});
Object.defineProperty(exports, "useIdle", {
  enumerable: true,
  get: function () {
    return _useIdle.useIdle;
  }
});
Object.defineProperty(exports, "useInViewport", {
  enumerable: true,
  get: function () {
    return _useInViewport.useInViewport;
  }
});
Object.defineProperty(exports, "useInputState", {
  enumerable: true,
  get: function () {
    return _useInputState.useInputState;
  }
});
Object.defineProperty(exports, "useIntersection", {
  enumerable: true,
  get: function () {
    return _useIntersection.useIntersection;
  }
});
Object.defineProperty(exports, "useInterval", {
  enumerable: true,
  get: function () {
    return _useInterval.useInterval;
  }
});
Object.defineProperty(exports, "useIsFirstRender", {
  enumerable: true,
  get: function () {
    return _useIsFirstRender.useIsFirstRender;
  }
});
Object.defineProperty(exports, "useIsomorphicEffect", {
  enumerable: true,
  get: function () {
    return _useIsomorphicEffect.useIsomorphicEffect;
  }
});
Object.defineProperty(exports, "useListState", {
  enumerable: true,
  get: function () {
    return _useListState.useListState;
  }
});
Object.defineProperty(exports, "useLocalStorage", {
  enumerable: true,
  get: function () {
    return _useLocalStorage.useLocalStorage;
  }
});
Object.defineProperty(exports, "useLogger", {
  enumerable: true,
  get: function () {
    return _useLogger.useLogger;
  }
});
Object.defineProperty(exports, "useMap", {
  enumerable: true,
  get: function () {
    return _useMap.useMap;
  }
});
Object.defineProperty(exports, "useMediaQuery", {
  enumerable: true,
  get: function () {
    return _useMediaQuery.useMediaQuery;
  }
});
Object.defineProperty(exports, "useMergedRef", {
  enumerable: true,
  get: function () {
    return _useMergedRef.useMergedRef;
  }
});
Object.defineProperty(exports, "useMounted", {
  enumerable: true,
  get: function () {
    return _useMounted.useMounted;
  }
});
Object.defineProperty(exports, "useMouse", {
  enumerable: true,
  get: function () {
    return _useMouse.useMouse;
  }
});
Object.defineProperty(exports, "useMove", {
  enumerable: true,
  get: function () {
    return _useMove.useMove;
  }
});
Object.defineProperty(exports, "useMutationObserver", {
  enumerable: true,
  get: function () {
    return _useMutationObserver.useMutationObserver;
  }
});
Object.defineProperty(exports, "useNetwork", {
  enumerable: true,
  get: function () {
    return _useNetwork.useNetwork;
  }
});
Object.defineProperty(exports, "useOrientation", {
  enumerable: true,
  get: function () {
    return _useOrientation.useOrientation;
  }
});
Object.defineProperty(exports, "useOs", {
  enumerable: true,
  get: function () {
    return _useOs.useOs;
  }
});
Object.defineProperty(exports, "usePageLeave", {
  enumerable: true,
  get: function () {
    return _usePageLeave.usePageLeave;
  }
});
Object.defineProperty(exports, "usePagination", {
  enumerable: true,
  get: function () {
    return _usePagination.usePagination;
  }
});
Object.defineProperty(exports, "usePrevious", {
  enumerable: true,
  get: function () {
    return _usePrevious.usePrevious;
  }
});
Object.defineProperty(exports, "useQueue", {
  enumerable: true,
  get: function () {
    return _useQueue.useQueue;
  }
});
Object.defineProperty(exports, "useReducedMotion", {
  enumerable: true,
  get: function () {
    return _useReducedMotion.useReducedMotion;
  }
});
Object.defineProperty(exports, "useResizeObserver", {
  enumerable: true,
  get: function () {
    return _useResizeObserver.useResizeObserver;
  }
});
Object.defineProperty(exports, "useScrollIntoView", {
  enumerable: true,
  get: function () {
    return _useScrollIntoView.useScrollIntoView;
  }
});
Object.defineProperty(exports, "useSessionStorage", {
  enumerable: true,
  get: function () {
    return _useSessionStorage.useSessionStorage;
  }
});
Object.defineProperty(exports, "useSet", {
  enumerable: true,
  get: function () {
    return _useSet.useSet;
  }
});
Object.defineProperty(exports, "useSetState", {
  enumerable: true,
  get: function () {
    return _useSetState.useSetState;
  }
});
Object.defineProperty(exports, "useShallowEffect", {
  enumerable: true,
  get: function () {
    return _useShallowEffect.useShallowEffect;
  }
});
Object.defineProperty(exports, "useStateHistory", {
  enumerable: true,
  get: function () {
    return _useStateHistory.useStateHistory;
  }
});
Object.defineProperty(exports, "useTextSelection", {
  enumerable: true,
  get: function () {
    return _useTextSelection.useTextSelection;
  }
});
Object.defineProperty(exports, "useThrottledCallback", {
  enumerable: true,
  get: function () {
    return _useThrottledCallback.useThrottledCallback;
  }
});
Object.defineProperty(exports, "useThrottledState", {
  enumerable: true,
  get: function () {
    return _useThrottledState.useThrottledState;
  }
});
Object.defineProperty(exports, "useThrottledValue", {
  enumerable: true,
  get: function () {
    return _useThrottledValue.useThrottledValue;
  }
});
Object.defineProperty(exports, "useTimeout", {
  enumerable: true,
  get: function () {
    return _useTimeout.useTimeout;
  }
});
Object.defineProperty(exports, "useToggle", {
  enumerable: true,
  get: function () {
    return _useToggle.useToggle;
  }
});
Object.defineProperty(exports, "useUncontrolled", {
  enumerable: true,
  get: function () {
    return _useUncontrolled.useUncontrolled;
  }
});
Object.defineProperty(exports, "useValidatedState", {
  enumerable: true,
  get: function () {
    return _useValidatedState.useValidatedState;
  }
});
Object.defineProperty(exports, "useViewportSize", {
  enumerable: true,
  get: function () {
    return _useViewportSize.useViewportSize;
  }
});
Object.defineProperty(exports, "useWindowEvent", {
  enumerable: true,
  get: function () {
    return _useWindowEvent.useWindowEvent;
  }
});
Object.defineProperty(exports, "useWindowScroll", {
  enumerable: true,
  get: function () {
    return _useWindowScroll.useWindowScroll;
  }
});
var _index = require("./utils/index.js");
Object.keys(_index).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _index[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _index[key];
    }
  });
});
var _useCallbackRef = require("./use-callback-ref/use-callback-ref");
var _useDebouncedCallback = require("./use-debounced-callback/use-debounced-callback");
var _useClickOutside = require("./use-click-outside/use-click-outside");
var _useClipboard = require("./use-clipboard/use-clipboard");
var _useColorScheme = require("./use-color-scheme/use-color-scheme");
var _useCounter = require("./use-counter/use-counter");
var _useDebouncedState = require("./use-debounced-state/use-debounced-state");
var _useDebouncedValue = require("./use-debounced-value/use-debounced-value");
var _useDocumentTitle = require("./use-document-title/use-document-title");
var _useDocumentVisibility = require("./use-document-visibility/use-document-visibility");
var _useFocusReturn = require("./use-focus-return/use-focus-return");
var _useDidUpdate = require("./use-did-update/use-did-update");
var _useFocusTrap = require("./use-focus-trap/use-focus-trap");
var _useForceUpdate = require("./use-force-update/use-force-update");
var _useId = require("./use-id/use-id");
var _useIdle = require("./use-idle/use-idle");
var _useInterval = require("./use-interval/use-interval");
var _useIsomorphicEffect = require("./use-isomorphic-effect/use-isomorphic-effect");
var _useListState = require("./use-list-state/use-list-state");
var _useLocalStorage = require("./use-local-storage/use-local-storage");
var _useSessionStorage = require("./use-session-storage/use-session-storage");
var _useMediaQuery = require("./use-media-query/use-media-query");
var _useMergedRef = require("./use-merged-ref/use-merged-ref");
var _useMouse = require("./use-mouse/use-mouse");
var _useMove = require("./use-move/use-move");
var _usePagination = require("./use-pagination/use-pagination");
var _useQueue = require("./use-queue/use-queue");
var _usePageLeave = require("./use-page-leave/use-page-leave");
var _useReducedMotion = require("./use-reduced-motion/use-reduced-motion");
var _useScrollIntoView = require("./use-scroll-into-view/use-scroll-into-view");
var _useResizeObserver = require("./use-resize-observer/use-resize-observer");
var _useShallowEffect = require("./use-shallow-effect/use-shallow-effect");
var _useToggle = require("./use-toggle/use-toggle");
var _useUncontrolled = require("./use-uncontrolled/use-uncontrolled");
var _useViewportSize = require("./use-viewport-size/use-viewport-size");
var _useWindowEvent = require("./use-window-event/use-window-event");
var _useWindowScroll = require("./use-window-scroll/use-window-scroll");
var _useIntersection = require("./use-intersection/use-intersection");
var _useHash = require("./use-hash/use-hash");
var _useHotkeys = require("./use-hotkeys/use-hotkeys");
var _useFullscreen = require("./use-fullscreen/use-fullscreen");
var _useLogger = require("./use-logger/use-logger");
var _useHover = require("./use-hover/use-hover");
var _useValidatedState = require("./use-validated-state/use-validated-state");
var _useOs = require("./use-os/use-os");
var _useSetState = require("./use-set-state/use-set-state");
var _useInputState = require("./use-input-state/use-input-state");
var _useEventListener = require("./use-event-listener/use-event-listener");
var _useDisclosure = require("./use-disclosure/use-disclosure");
var _useFocusWithin = require("./use-focus-within/use-focus-within");
var _useNetwork = require("./use-network/use-network");
var _useTimeout = require("./use-timeout/use-timeout");
var _useTextSelection = require("./use-text-selection/use-text-selection");
var _usePrevious = require("./use-previous/use-previous");
var _useFavicon = require("./use-favicon/use-favicon");
var _useHeadroom = require("./use-headroom/use-headroom");
var _useEyeDropper = require("./use-eye-dropper/use-eye-dropper");
var _useInViewport = require("./use-in-viewport/use-in-viewport");
var _useMutationObserver = require("./use-mutation-observer/use-mutation-observer");
var _useMounted = require("./use-mounted/use-mounted");
var _useStateHistory = require("./use-state-history/use-state-history");
var _useMap = require("./use-map/use-map");
var _useSet = require("./use-set/use-set");
var _useThrottledCallback = require("./use-throttled-callback/use-throttled-callback");
var _useThrottledState = require("./use-throttled-state/use-throttled-state");
var _useThrottledValue = require("./use-throttled-value/use-throttled-value");
var _useIsFirstRender = require("./use-is-first-render/use-is-first-render");
var _useOrientation = require("./use-orientation/use-orientation");
var _useFetch = require("./use-fetch/use-fetch");
