// @mantine/core/lib/core/MantineProvider/color-functions/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "alpha", {
  enumerable: true,
  get: function () {
    return _rgba.alpha;
  }
});
Object.defineProperty(exports, "colorsTuple", {
  enumerable: true,
  get: function () {
    return _colorsTuple.colorsTuple;
  }
});
Object.defineProperty(exports, "darken", {
  enumerable: true,
  get: function () {
    return _darken.darken;
  }
});
Object.defineProperty(exports, "defaultVariantColorsResolver", {
  enumerable: true,
  get: function () {
    return _defaultVariantColorsResolver.defaultVariantColorsResolver;
  }
});
Object.defineProperty(exports, "getAutoContrastValue", {
  enumerable: true,
  get: function () {
    return _getAutoContrastValue.getAutoContrastValue;
  }
});
Object.defineProperty(exports, "getContrastColor", {
  enumerable: true,
  get: function () {
    return _getContrastColor.getContrastColor;
  }
});
Object.defineProperty(exports, "getGradient", {
  enumerable: true,
  get: function () {
    return _getGradient.getGradient;
  }
});
Object.defineProperty(exports, "getPrimaryContrastColor", {
  enumerable: true,
  get: function () {
    return _getContrastColor.getPrimaryContrastColor;
  }
});
Object.defineProperty(exports, "getPrimaryShade", {
  enumerable: true,
  get: function () {
    return _getPrimaryShade.getPrimaryShade;
  }
});
Object.defineProperty(exports, "getThemeColor", {
  enumerable: true,
  get: function () {
    return _getThemeColor.getThemeColor;
  }
});
Object.defineProperty(exports, "isLightColor", {
  enumerable: true,
  get: function () {
    return _luminance.isLightColor;
  }
});
Object.defineProperty(exports, "lighten", {
  enumerable: true,
  get: function () {
    return _lighten.lighten;
  }
});
Object.defineProperty(exports, "luminance", {
  enumerable: true,
  get: function () {
    return _luminance.luminance;
  }
});
Object.defineProperty(exports, "parseThemeColor", {
  enumerable: true,
  get: function () {
    return _parseThemeColor.parseThemeColor;
  }
});
Object.defineProperty(exports, "rgba", {
  enumerable: true,
  get: function () {
    return _rgba.rgba;
  }
});
Object.defineProperty(exports, "toRgba", {
  enumerable: true,
  get: function () {
    return _toRgba.toRgba;
  }
});
var _getPrimaryShade = require("./get-primary-shade/get-primary-shade");
var _parseThemeColor = require("./parse-theme-color/parse-theme-color");
var _getThemeColor = require("./get-theme-color/get-theme-color");
var _defaultVariantColorsResolver = require("./default-variant-colors-resolver/default-variant-colors-resolver");
var _getGradient = require("./get-gradient/get-gradient");
var _toRgba = require("./to-rgba/to-rgba");
var _rgba = require("./rgba/rgba");
var _darken = require("./darken/darken");
var _lighten = require("./lighten/lighten");
var _luminance = require("./luminance/luminance");
var _getContrastColor = require("./get-contrast-color/get-contrast-color");
var _getAutoContrastValue = require("./get-auto-contrast-value/get-auto-contrast-value");
var _colorsTuple = require("./colors-tuple/colors-tuple");
