// @mantine/core/lib/components/Pagination/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Pagination", {
  enumerable: true,
  get: function () {
    return _Pagination.Pagination;
  }
});
Object.defineProperty(exports, "PaginationControl", {
  enumerable: true,
  get: function () {
    return _PaginationControl.PaginationControl;
  }
});
Object.defineProperty(exports, "PaginationDots", {
  enumerable: true,
  get: function () {
    return _PaginationDots.PaginationDots;
  }
});
Object.defineProperty(exports, "PaginationFirst", {
  enumerable: true,
  get: function () {
    return _PaginationEdges.PaginationFirst;
  }
});
Object.defineProperty(exports, "PaginationItems", {
  enumerable: true,
  get: function () {
    return _PaginationItems.PaginationItems;
  }
});
Object.defineProperty(exports, "PaginationLast", {
  enumerable: true,
  get: function () {
    return _PaginationEdges.PaginationLast;
  }
});
Object.defineProperty(exports, "PaginationNext", {
  enumerable: true,
  get: function () {
    return _PaginationEdges.PaginationNext;
  }
});
Object.defineProperty(exports, "PaginationPrevious", {
  enumerable: true,
  get: function () {
    return _PaginationEdges.PaginationPrevious;
  }
});
Object.defineProperty(exports, "PaginationRoot", {
  enumerable: true,
  get: function () {
    return _PaginationRoot.PaginationRoot;
  }
});
var _Pagination = require("./Pagination");
var _PaginationControl = require("./PaginationControl/PaginationControl");
var _PaginationDots = require("./PaginationDots/PaginationDots");
var _PaginationEdges = require("./PaginationEdges/PaginationEdges");
var _PaginationItems = require("./PaginationItems/PaginationItems");
var _PaginationRoot = require("./PaginationRoot/PaginationRoot");
