// @mantine/core/lib/components/Table/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return _Table.Table;
  }
});
Object.defineProperty(exports, "TableCaption", {
  enumerable: true,
  get: function () {
    return _Table2.TableCaption;
  }
});
Object.defineProperty(exports, "TableScrollContainer", {
  enumerable: true,
  get: function () {
    return _TableScrollContainer.TableScrollContainer;
  }
});
Object.defineProperty(exports, "TableTbody", {
  enumerable: true,
  get: function () {
    return _Table2.TableTbody;
  }
});
Object.defineProperty(exports, "TableTd", {
  enumerable: true,
  get: function () {
    return _Table2.TableTd;
  }
});
Object.defineProperty(exports, "TableTfoot", {
  enumerable: true,
  get: function () {
    return _Table2.TableTfoot;
  }
});
Object.defineProperty(exports, "TableTh", {
  enumerable: true,
  get: function () {
    return _Table2.TableTh;
  }
});
Object.defineProperty(exports, "TableThead", {
  enumerable: true,
  get: function () {
    return _Table2.TableThead;
  }
});
Object.defineProperty(exports, "TableTr", {
  enumerable: true,
  get: function () {
    return _Table2.TableTr;
  }
});
var _Table = require("./Table");
var _Table2 = require("./Table.components");
var _TableScrollContainer = require("./TableScrollContainer");
