// @floating-ui/react-dom/dist/floating-ui.react-dom.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AlignedPlacement", {
  enumerable: true,
  get: function () {
    return _dom.AlignedPlacement;
  }
});
Object.defineProperty(exports, "Alignment", {
  enumerable: true,
  get: function () {
    return _dom.Alignment;
  }
});
Object.defineProperty(exports, "AutoPlacementOptions", {
  enumerable: true,
  get: function () {
    return _dom.AutoPlacementOptions;
  }
});
Object.defineProperty(exports, "AutoUpdateOptions", {
  enumerable: true,
  get: function () {
    return _dom.AutoUpdateOptions;
  }
});
Object.defineProperty(exports, "Axis", {
  enumerable: true,
  get: function () {
    return _dom.Axis;
  }
});
Object.defineProperty(exports, "Boundary", {
  enumerable: true,
  get: function () {
    return _dom.Boundary;
  }
});
Object.defineProperty(exports, "ClientRectObject", {
  enumerable: true,
  get: function () {
    return _dom.ClientRectObject;
  }
});
Object.defineProperty(exports, "ComputePositionConfig", {
  enumerable: true,
  get: function () {
    return _dom.ComputePositionConfig;
  }
});
Object.defineProperty(exports, "ComputePositionReturn", {
  enumerable: true,
  get: function () {
    return _dom.ComputePositionReturn;
  }
});
Object.defineProperty(exports, "Coords", {
  enumerable: true,
  get: function () {
    return _dom.Coords;
  }
});
Object.defineProperty(exports, "DetectOverflowOptions", {
  enumerable: true,
  get: function () {
    return _dom.DetectOverflowOptions;
  }
});
Object.defineProperty(exports, "Dimensions", {
  enumerable: true,
  get: function () {
    return _dom.Dimensions;
  }
});
Object.defineProperty(exports, "ElementContext", {
  enumerable: true,
  get: function () {
    return _dom.ElementContext;
  }
});
Object.defineProperty(exports, "ElementRects", {
  enumerable: true,
  get: function () {
    return _dom.ElementRects;
  }
});
Object.defineProperty(exports, "Elements", {
  enumerable: true,
  get: function () {
    return _dom.Elements;
  }
});
Object.defineProperty(exports, "FlipOptions", {
  enumerable: true,
  get: function () {
    return _dom.FlipOptions;
  }
});
Object.defineProperty(exports, "FloatingElement", {
  enumerable: true,
  get: function () {
    return _dom.FloatingElement;
  }
});
Object.defineProperty(exports, "HideOptions", {
  enumerable: true,
  get: function () {
    return _dom.HideOptions;
  }
});
Object.defineProperty(exports, "InlineOptions", {
  enumerable: true,
  get: function () {
    return _dom.InlineOptions;
  }
});
Object.defineProperty(exports, "Length", {
  enumerable: true,
  get: function () {
    return _dom.Length;
  }
});
Object.defineProperty(exports, "Middleware", {
  enumerable: true,
  get: function () {
    return _dom.Middleware;
  }
});
Object.defineProperty(exports, "MiddlewareArguments", {
  enumerable: true,
  get: function () {
    return _dom.MiddlewareArguments;
  }
});
Object.defineProperty(exports, "MiddlewareData", {
  enumerable: true,
  get: function () {
    return _dom.MiddlewareData;
  }
});
Object.defineProperty(exports, "MiddlewareReturn", {
  enumerable: true,
  get: function () {
    return _dom.MiddlewareReturn;
  }
});
Object.defineProperty(exports, "MiddlewareState", {
  enumerable: true,
  get: function () {
    return _dom.MiddlewareState;
  }
});
Object.defineProperty(exports, "NodeScroll", {
  enumerable: true,
  get: function () {
    return _dom.NodeScroll;
  }
});
Object.defineProperty(exports, "OffsetOptions", {
  enumerable: true,
  get: function () {
    return _dom.OffsetOptions;
  }
});
Object.defineProperty(exports, "Padding", {
  enumerable: true,
  get: function () {
    return _dom.Padding;
  }
});
Object.defineProperty(exports, "Placement", {
  enumerable: true,
  get: function () {
    return _dom.Placement;
  }
});
Object.defineProperty(exports, "Platform", {
  enumerable: true,
  get: function () {
    return _dom.Platform;
  }
});
Object.defineProperty(exports, "Rect", {
  enumerable: true,
  get: function () {
    return _dom.Rect;
  }
});
Object.defineProperty(exports, "ReferenceElement", {
  enumerable: true,
  get: function () {
    return _dom.ReferenceElement;
  }
});
Object.defineProperty(exports, "RootBoundary", {
  enumerable: true,
  get: function () {
    return _dom.RootBoundary;
  }
});
Object.defineProperty(exports, "ShiftOptions", {
  enumerable: true,
  get: function () {
    return _dom.ShiftOptions;
  }
});
Object.defineProperty(exports, "Side", {
  enumerable: true,
  get: function () {
    return _dom.Side;
  }
});
Object.defineProperty(exports, "SideObject", {
  enumerable: true,
  get: function () {
    return _dom.SideObject;
  }
});
Object.defineProperty(exports, "SizeOptions", {
  enumerable: true,
  get: function () {
    return _dom.SizeOptions;
  }
});
Object.defineProperty(exports, "Strategy", {
  enumerable: true,
  get: function () {
    return _dom.Strategy;
  }
});
Object.defineProperty(exports, "VirtualElement", {
  enumerable: true,
  get: function () {
    return _dom.VirtualElement;
  }
});
Object.defineProperty(exports, "autoPlacement", {
  enumerable: true,
  get: function () {
    return _dom.autoPlacement;
  }
});
Object.defineProperty(exports, "autoUpdate", {
  enumerable: true,
  get: function () {
    return _dom.autoUpdate;
  }
});
Object.defineProperty(exports, "computePosition", {
  enumerable: true,
  get: function () {
    return _dom.computePosition;
  }
});
Object.defineProperty(exports, "detectOverflow", {
  enumerable: true,
  get: function () {
    return _dom.detectOverflow;
  }
});
Object.defineProperty(exports, "flip", {
  enumerable: true,
  get: function () {
    return _dom.flip;
  }
});
Object.defineProperty(exports, "getOverflowAncestors", {
  enumerable: true,
  get: function () {
    return _dom.getOverflowAncestors;
  }
});
Object.defineProperty(exports, "hide", {
  enumerable: true,
  get: function () {
    return _dom.hide;
  }
});
Object.defineProperty(exports, "inline", {
  enumerable: true,
  get: function () {
    return _dom.inline;
  }
});
Object.defineProperty(exports, "limitShift", {
  enumerable: true,
  get: function () {
    return _dom.limitShift;
  }
});
Object.defineProperty(exports, "offset", {
  enumerable: true,
  get: function () {
    return _dom.offset;
  }
});
Object.defineProperty(exports, "platform", {
  enumerable: true,
  get: function () {
    return _dom.platform;
  }
});
Object.defineProperty(exports, "shift", {
  enumerable: true,
  get: function () {
    return _dom.shift;
  }
});
Object.defineProperty(exports, "size", {
  enumerable: true,
  get: function () {
    return _dom.size;
  }
});
var _dom = require("@floating-ui/dom");
