// @mantine/core/lib/components/ScrollArea/utils/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "addUnlinkedScrollListener", {
  enumerable: true,
  get: function () {
    return _addUnlinkedScrollListener.addUnlinkedScrollListener;
  }
});
Object.defineProperty(exports, "composeEventHandlers", {
  enumerable: true,
  get: function () {
    return _composeEventHandlers.composeEventHandlers;
  }
});
Object.defineProperty(exports, "getScrollPositionFromPointer", {
  enumerable: true,
  get: function () {
    return _getScrollPositionFromPointer.getScrollPositionFromPointer;
  }
});
Object.defineProperty(exports, "getThumbOffsetFromScroll", {
  enumerable: true,
  get: function () {
    return _getThumbOffsetFromScroll.getThumbOffsetFromScroll;
  }
});
Object.defineProperty(exports, "getThumbRatio", {
  enumerable: true,
  get: function () {
    return _getThumbRatio.getThumbRatio;
  }
});
Object.defineProperty(exports, "getThumbSize", {
  enumerable: true,
  get: function () {
    return _getThumbSize.getThumbSize;
  }
});
Object.defineProperty(exports, "isScrollingWithinScrollbarBounds", {
  enumerable: true,
  get: function () {
    return _isScrollingWithinScrollbarBounds.isScrollingWithinScrollbarBounds;
  }
});
Object.defineProperty(exports, "toInt", {
  enumerable: true,
  get: function () {
    return _toInt.toInt;
  }
});
var _isScrollingWithinScrollbarBounds = require("./is-scrolling-within-scrollbar-bounds");
var _getThumbRatio = require("./get-thumb-ratio");
var _getThumbSize = require("./get-thumb-size");
var _composeEventHandlers = require("./compose-event-handlers");
var _toInt = require("./to-int");
var _getThumbOffsetFromScroll = require("./get-thumb-offset-from-scroll");
var _getScrollPositionFromPointer = require("./get-scroll-position-from-pointer");
var _addUnlinkedScrollListener = require("./add-unlinked-scroll-listener");
