// @mantine/core/lib/components/Modal/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Modal", {
  enumerable: true,
  get: function () {
    return _Modal.Modal;
  }
});
Object.defineProperty(exports, "ModalBody", {
  enumerable: true,
  get: function () {
    return _ModalBody.ModalBody;
  }
});
Object.defineProperty(exports, "ModalCloseButton", {
  enumerable: true,
  get: function () {
    return _ModalCloseButton.ModalCloseButton;
  }
});
Object.defineProperty(exports, "ModalContent", {
  enumerable: true,
  get: function () {
    return _ModalContent.ModalContent;
  }
});
Object.defineProperty(exports, "ModalHeader", {
  enumerable: true,
  get: function () {
    return _ModalHeader.ModalHeader;
  }
});
Object.defineProperty(exports, "ModalOverlay", {
  enumerable: true,
  get: function () {
    return _ModalOverlay.ModalOverlay;
  }
});
Object.defineProperty(exports, "ModalRoot", {
  enumerable: true,
  get: function () {
    return _ModalRoot.ModalRoot;
  }
});
Object.defineProperty(exports, "ModalTitle", {
  enumerable: true,
  get: function () {
    return _ModalTitle.ModalTitle;
  }
});
var _Modal = require("./Modal");
var _ModalRoot = require("./ModalRoot");
var _ModalBody = require("./ModalBody");
var _ModalCloseButton = require("./ModalCloseButton");
var _ModalContent = require("./ModalContent");
var _ModalHeader = require("./ModalHeader");
var _ModalOverlay = require("./ModalOverlay");
var _ModalTitle = require("./ModalTitle");
