// @mantine/core/lib/components/ModalBase/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ModalBase", {
  enumerable: true,
  get: function () {
    return _ModalBase.ModalBase;
  }
});
Object.defineProperty(exports, "ModalBaseBody", {
  enumerable: true,
  get: function () {
    return _ModalBaseBody.ModalBaseBody;
  }
});
Object.defineProperty(exports, "ModalBaseCloseButton", {
  enumerable: true,
  get: function () {
    return _ModalBaseCloseButton.ModalBaseCloseButton;
  }
});
Object.defineProperty(exports, "ModalBaseContent", {
  enumerable: true,
  get: function () {
    return _ModalBaseContent.ModalBaseContent;
  }
});
Object.defineProperty(exports, "ModalBaseHeader", {
  enumerable: true,
  get: function () {
    return _ModalBaseHeader.ModalBaseHeader;
  }
});
Object.defineProperty(exports, "ModalBaseOverlay", {
  enumerable: true,
  get: function () {
    return _ModalBaseOverlay.ModalBaseOverlay;
  }
});
Object.defineProperty(exports, "ModalBaseTitle", {
  enumerable: true,
  get: function () {
    return _ModalBaseTitle.ModalBaseTitle;
  }
});
Object.defineProperty(exports, "NativeScrollArea", {
  enumerable: true,
  get: function () {
    return _NativeScrollArea.NativeScrollArea;
  }
});
var _ModalBase = require("./ModalBase");
var _ModalBaseBody = require("./ModalBaseBody");
var _ModalBaseCloseButton = require("./ModalBaseCloseButton");
var _ModalBaseContent = require("./ModalBaseContent");
var _ModalBaseHeader = require("./ModalBaseHeader");
var _ModalBaseOverlay = require("./ModalBaseOverlay");
var _ModalBaseTitle = require("./ModalBaseTitle");
var _NativeScrollArea = require("./NativeScrollArea");
