// @mantine/core/lib/core/MantineProvider/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  DEFAULT_THEME: true,
  validateMantineTheme: true,
  mergeMantineTheme: true,
  MantineProvider: true,
  HeadlessMantineProvider: true,
  MantineContext: true,
  useMantineContext: true,
  useMantineClassNamesPrefix: true,
  useMantineStyleNonce: true,
  useMantineCssVariablesResolver: true,
  useMantineWithStaticClasses: true,
  useMantineIsHeadless: true,
  useMantineTheme: true,
  useSafeMantineTheme: true,
  MantineThemeContext: true,
  MantineThemeProvider: true,
  useProps: true,
  convertCssVariables: true,
  createTheme: true,
  mergeThemeOverrides: true,
  defaultCssVariablesResolver: true,
  virtualColor: true,
  getCSSColorVariables: true,
  isVirtualColor: true
};
Object.defineProperty(exports, "DEFAULT_THEME", {
  enumerable: true,
  get: function () {
    return _defaultTheme.DEFAULT_THEME;
  }
});
Object.defineProperty(exports, "HeadlessMantineProvider", {
  enumerable: true,
  get: function () {
    return _MantineProvider.HeadlessMantineProvider;
  }
});
Object.defineProperty(exports, "MantineContext", {
  enumerable: true,
  get: function () {
    return _Mantine.MantineContext;
  }
});
Object.defineProperty(exports, "MantineProvider", {
  enumerable: true,
  get: function () {
    return _MantineProvider.MantineProvider;
  }
});
Object.defineProperty(exports, "MantineThemeContext", {
  enumerable: true,
  get: function () {
    return _MantineThemeProvider.MantineThemeContext;
  }
});
Object.defineProperty(exports, "MantineThemeProvider", {
  enumerable: true,
  get: function () {
    return _MantineThemeProvider.MantineThemeProvider;
  }
});
Object.defineProperty(exports, "convertCssVariables", {
  enumerable: true,
  get: function () {
    return _convertCssVariables.convertCssVariables;
  }
});
Object.defineProperty(exports, "createTheme", {
  enumerable: true,
  get: function () {
    return _createTheme.createTheme;
  }
});
Object.defineProperty(exports, "defaultCssVariablesResolver", {
  enumerable: true,
  get: function () {
    return _defaultCssVariablesResolver.defaultCssVariablesResolver;
  }
});
Object.defineProperty(exports, "getCSSColorVariables", {
  enumerable: true,
  get: function () {
    return _MantineCssVariables.getCSSColorVariables;
  }
});
Object.defineProperty(exports, "isVirtualColor", {
  enumerable: true,
  get: function () {
    return _MantineCssVariables.isVirtualColor;
  }
});
Object.defineProperty(exports, "mergeMantineTheme", {
  enumerable: true,
  get: function () {
    return _mergeMantineTheme.mergeMantineTheme;
  }
});
Object.defineProperty(exports, "mergeThemeOverrides", {
  enumerable: true,
  get: function () {
    return _mergeThemeOverrides.mergeThemeOverrides;
  }
});
Object.defineProperty(exports, "useMantineClassNamesPrefix", {
  enumerable: true,
  get: function () {
    return _Mantine.useMantineClassNamesPrefix;
  }
});
Object.defineProperty(exports, "useMantineContext", {
  enumerable: true,
  get: function () {
    return _Mantine.useMantineContext;
  }
});
Object.defineProperty(exports, "useMantineCssVariablesResolver", {
  enumerable: true,
  get: function () {
    return _Mantine.useMantineCssVariablesResolver;
  }
});
Object.defineProperty(exports, "useMantineIsHeadless", {
  enumerable: true,
  get: function () {
    return _Mantine.useMantineIsHeadless;
  }
});
Object.defineProperty(exports, "useMantineStyleNonce", {
  enumerable: true,
  get: function () {
    return _Mantine.useMantineStyleNonce;
  }
});
Object.defineProperty(exports, "useMantineTheme", {
  enumerable: true,
  get: function () {
    return _MantineThemeProvider.useMantineTheme;
  }
});
Object.defineProperty(exports, "useMantineWithStaticClasses", {
  enumerable: true,
  get: function () {
    return _Mantine.useMantineWithStaticClasses;
  }
});
Object.defineProperty(exports, "useProps", {
  enumerable: true,
  get: function () {
    return _useProps.useProps;
  }
});
Object.defineProperty(exports, "useSafeMantineTheme", {
  enumerable: true,
  get: function () {
    return _MantineThemeProvider.useSafeMantineTheme;
  }
});
Object.defineProperty(exports, "validateMantineTheme", {
  enumerable: true,
  get: function () {
    return _mergeMantineTheme.validateMantineTheme;
  }
});
Object.defineProperty(exports, "virtualColor", {
  enumerable: true,
  get: function () {
    return _MantineCssVariables.virtualColor;
  }
});
var _theme = require("./theme.types");
Object.keys(_theme).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _theme[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _theme[key];
    }
  });
});
var _colorSchemeManagers = require("./color-scheme-managers");
Object.keys(_colorSchemeManagers).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _colorSchemeManagers[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _colorSchemeManagers[key];
    }
  });
});
var _colorFunctions = require("./color-functions");
Object.keys(_colorFunctions).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _colorFunctions[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _colorFunctions[key];
    }
  });
});
var _useMantineColorScheme = require("./use-mantine-color-scheme");
Object.keys(_useMantineColorScheme).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _useMantineColorScheme[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _useMantineColorScheme[key];
    }
  });
});
var _ColorSchemeScript = require("./ColorSchemeScript");
Object.keys(_ColorSchemeScript).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ColorSchemeScript[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ColorSchemeScript[key];
    }
  });
});
var _defaultTheme = require("./default-theme");
var _mergeMantineTheme = require("./merge-mantine-theme");
var _MantineProvider = require("./MantineProvider");
var _Mantine = require("./Mantine.context");
var _MantineThemeProvider = require("./MantineThemeProvider");
var _useProps = require("./use-props/use-props");
var _convertCssVariables = require("./convert-css-variables");
var _createTheme = require("./create-theme/create-theme");
var _mergeThemeOverrides = require("./merge-theme-overrides/merge-theme-overrides");
var _defaultCssVariablesResolver = require("./MantineCssVariables/default-css-variables-resolver");
var _MantineCssVariables = require("./MantineCssVariables");
