// @mantine/core/lib/components/Menu/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Menu", {
  enumerable: true,
  get: function () {
    return _Menu.Menu;
  }
});
Object.defineProperty(exports, "MenuDivider", {
  enumerable: true,
  get: function () {
    return _MenuDivider.MenuDivider;
  }
});
Object.defineProperty(exports, "MenuDropdown", {
  enumerable: true,
  get: function () {
    return _MenuDropdown.MenuDropdown;
  }
});
Object.defineProperty(exports, "MenuItem", {
  enumerable: true,
  get: function () {
    return _MenuItem.MenuItem;
  }
});
Object.defineProperty(exports, "MenuLabel", {
  enumerable: true,
  get: function () {
    return _MenuLabel.MenuLabel;
  }
});
Object.defineProperty(exports, "MenuTarget", {
  enumerable: true,
  get: function () {
    return _MenuTarget.MenuTarget;
  }
});
var _Menu = require("./Menu");
var _MenuDivider = require("./MenuDivider/MenuDivider");
var _MenuItem = require("./MenuItem/MenuItem");
var _MenuLabel = require("./MenuLabel/MenuLabel");
var _MenuDropdown = require("./MenuDropdown/MenuDropdown");
var _MenuTarget = require("./MenuTarget/MenuTarget");
