// @mantine/core/lib/components/Input/index.d.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function () {
    return _Input.Input;
  }
});
Object.defineProperty(exports, "InputDescription", {
  enumerable: true,
  get: function () {
    return _InputDescription.InputDescription;
  }
});
Object.defineProperty(exports, "InputError", {
  enumerable: true,
  get: function () {
    return _InputError.InputError;
  }
});
Object.defineProperty(exports, "InputLabel", {
  enumerable: true,
  get: function () {
    return _InputLabel.InputLabel;
  }
});
Object.defineProperty(exports, "InputPlaceholder", {
  enumerable: true,
  get: function () {
    return _InputPlaceholder.InputPlaceholder;
  }
});
Object.defineProperty(exports, "InputWrapper", {
  enumerable: true,
  get: function () {
    return _InputWrapper.InputWrapper;
  }
});
Object.defineProperty(exports, "useInputProps", {
  enumerable: true,
  get: function () {
    return _useInputProps.useInputProps;
  }
});
Object.defineProperty(exports, "useInputWrapperContext", {
  enumerable: true,
  get: function () {
    return _InputWrapper2.useInputWrapperContext;
  }
});
var _Input = require("./Input");
var _InputWrapper = require("./InputWrapper/InputWrapper");
var _InputDescription = require("./InputDescription/InputDescription");
var _InputError = require("./InputError/InputError");
var _InputLabel = require("./InputLabel/InputLabel");
var _InputPlaceholder = require("./InputPlaceholder/InputPlaceholder");
var _useInputProps = require("./use-input-props");
var _InputWrapper2 = require("./InputWrapper.context");
